packages/wrapped-keys/src/lib/litActions.ts

```diff
@@ -1,146 +1,193 @@
-export const signWithEthereumEncryptedKeyLitAction = `
-(async () => {
-    // TODO!: Remove ALL the console.log statements
-    console.log('unsignedTransaction');
-    console.log(unsignedTransaction);
+// @ts-nocheck
 
-    if (!unsignedTransaction.toAddress) {
-        Lit.Actions.setResponse({ response: 'Error: Missing required field: toAddress' });
-        return;
-    }
-
-    if (!unsignedTransaction.chain) {
-        Lit.Actions.setResponse({ response: 'Error: Missing required field: chain' });
-        return;
-    }
-
-    if (!unsignedTransaction.value) {
-        Lit.Actions.setResponse({ response: 'Error: Missing required field: value' });
-        return;
-    }
+export const generatePrivateKeyLitAction = `
+const LIT_PREFIX = 'lit_';
 
-    if (!unsignedTransaction.chainId) {
-        Lit.Actions.setResponse({ response: 'Error: Missing required field: chainId' });
-        return;
-    }
+(async () => {
+    const resp = await Lit.Actions.runOnce(
+        { waitForResponse: true, name: 'encryptedPrivateKey' },
+        async () => {
+            const wallet = ethers.Wallet.createRandom();
+            const privateKey = LIT_PREFIX + wallet.privateKey.toString();
+            let utf8Encode = new TextEncoder();
+            const to_encrypt = utf8Encode.encode(privateKey);
+
+            const { ciphertext, dataToEncryptHash } = await Lit.Actions.encrypt({
+                accessControlConditions,
+                to_encrypt,
+            });
+            return JSON.stringify({ ciphertext, dataToEncryptHash, publicKey: wallet.publicKey });
+        }
+    );
 
-    const decryptedPrivateKey = await Lit.Actions.decryptToSingleNode({
-        accessControlConditions,
-        ciphertext,
-        dataToEncryptHash,
-        chain: 'ethereum',
-        authSig: null,
+    // // TODO: Remove the below which is only for demonstrating the error
+    // const { ciphertext, dataToEncryptHash } = JSON.parse(resp);
+    // const decrypted = await Lit.Actions.decryptAndCombine({
+    //     accessControlConditions,
+    //     ciphertext,
+    //     dataToEncryptHash,
+    //     authSig: null,
+    //     chain: 'ethereum',
+    // });
+
+    // // TODO: Remove the below which is only for demonstrating the error
+    // console.log('accessControlConditions: ', accessControlConditions);
+    // console.log('ciphertext: ', ciphertext);
+    // console.log('dataToEncryptHash: ', dataToEncryptHash);
+    // console.log('decrypted: ', decrypted);
+
+    Lit.Actions.setResponse({
+        response: resp,
     });
+})();
+`;
 
-    console.log('decryptedPrivateKey');
-    console.log(decryptedPrivateKey);
-
-    if (!decryptedPrivateKey) { // Exit the nodes which don't have the decryptedData
-        return;
-    }
+const _signTransactionWithEthereumEncryptedKeyLitAction = (async () => {
+  const LIT_PREFIX = 'lit_';
```

Comment:
It's being used at multiple places lets pull this out from this function and reuse in the below functions?

Reply:
I tried pull it out and got the error
```
✖ 1. testEthereumBroadcastWrappedKeyWithFetchGasParams - Failed after 1 attempts (20303.62 ms)
❌Error: Error: Lit Action threw an unexpected error: {"message":"There was an error getting the signing shares from the nodes","errorCode":"unknown_error"}
Test Report: Some tests failed.
- testEthereumBroadcastWrappedKeyWithFetchGasParams (Failed in 20303.62 ms) - Error: Error: Lit Action threw an unexpected error: {"message":"There was an error getting the signing shares from the nodes","errorCode":"unknown_error"}
```
tried converting the LA functions to regular functions instead of arrow function incase it was a scoping issue, but same error

Reply:
Let's keep it like this then